core/commands/show_commit.py

```diff
@@ -8,18 +8,23 @@
 
 class GsShowCommitCommand(WindowCommand, GitCommand):
 
-    def run(self, commit_hash):
+    def run(self, commit_hash, show_file=False, filepath=None):
+        self.show_file = show_file
         repo_path = self.repo_path
         view = self.window.new_file()
         view.set_syntax_file("Packages/GitSavvy/syntax/show_commit.tmLanguage")
         view.settings().set("git_savvy.show_commit_view", True)
         view.settings().set("git_savvy.show_commit_view.commit", commit_hash)
+        view.settings().set("git_savvy.show_commit_view.filepath", filepath)
         view.settings().set("git_savvy.repo_path", repo_path)
         view.settings().set("word_wrap", False)
         view.settings().set("line_numbers", False)
         view.set_name(SHOW_COMMIT_TITLE.format(commit_hash[:7]))
         view.set_scratch(True)
-        view.run_command("gs_show_commit_initialize_view")
+        if self.show_file == True:
```

Comment:
No need for `if my_var == True:` - `if my_var:` is considered to be the idiomatic pattern.